spinnman/connections/scp_request_pipeline.py

```diff
@@ -200,16 +203,19 @@ def _single_retrieve(self, timeout):
 
         # Only process responses which have matching requests
         if seq in self._requests:
+            self._in_progress -= 1
             request_sent = self._requests[seq]
 
-            # If the response can be retried, retry it, as long as the
-            # timeout hasn't expired
-
-            if (result in RETRY_CODES and
-                    (time.time() - self._send_time[seq] <
-                        self._packet_timeout)):
-                self._connection.send(self._request_data[seq])
-                self._n_retry_code_resent += 1
+            # If the response can be retried, retry it
+            if (result in RETRY_CODES):
+                try:
+                    time.sleep(0.1)
```

Comment:
Do you want to sleep before _every_ send? Can we make this be so that it only happens on a retry?

Reply:
This only happens when the result is in the retry codes, so the first send won't go through this.